speedtest.py

```diff
@@ -0,0 +1,325 @@
+# Copyright (c) 2014 Jim Kemp <kemp.jim@gmail.com>
+# Copyright (c) 2017 Gene Liverman <gene@technicalissues.us>
+
+# Permission is hereby granted, free of charge, to any person
+# obtaining a copy of this software and associated documentation
+# files (the "Software"), to deal in the Software without
+# restriction, including without limitation the rights to use,
+# copy, modify, merge, publish, distribute, sublicense, and/or sell
+# copies of the Software, and to permit persons to whom the
+# Software is furnished to do so, subject to the following
+# conditions:
+#
+# The above copyright notice and this permission notice shall be
+# included in all copies or substantial portions of the Software.
+#
+# THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND,
+# EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES
+# OF MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND
+# NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT
+# HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY,
+# WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING
+# FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR
+# OTHER DEALINGS IN THE SOFTWARE.
+# END LICENSE
+
+""" Fetches weather reports from Dark Sky for displaying on a screen.
+    Plugin fetches network stats for displaying on a screen. """
+
+__version__ = "0.0.12"
+
+###############################################################################
+#   Raspberry Pi Weather Display
+#   Original By: Jim Kemp          10/25/2014
+#   Modified By: Gene Liverman    12/30/2017 & multiple times since
+###############################################################################
+
+# standard imports
+import datetime
+import os
+import syslog
+import time
+import subprocess
+import math
+import json
+import glob
+import shutil
+
+# third party imports
+import pygame
+
+# local imports
+from weather_rock_methods import *
+import plugin_configs.speedtest_config as speedtest_config
+
+
+class Speedtest:
+    def get_speedtest(self, last_update_time):
+
+        # Determine if this is the first time fetching new speedtest results.
+        initial = False
+        if last_update_time == 0:
+            initial = True
+
+        if initial:
+            # Make sure the speedtest directories exist.
+            os.makedirs(
+                os.path.join("/home/pi/PiWeatherRock/", "speedtest", "queue"),
+                exist_ok=True)
+            os.makedirs(
+                os.path.join(
+                    "/home/pi/PiWeatherRock/", "speedtest", "archive"),
+                exist_ok=True)
+
+        # Run speedtest on Pi and store the results in json format.
+        if speedtest_config.SPEEDTEST_ON_PI:
+            if ((time.time() -
+                 last_update_time) > speedtest_config.UPDATE_FREQ):
+                last_update_time = round(time.time())
+                syslog.syslog('Running speedtest on Pi.')
+                subprocess.Popen(["/bin/bash",
```

Comment:
Anyway to avoid a hardcoded path here a la how shell scripts use `#!/usr/bin/env bash`

Reply:
Looks like simply using `"sh",......` should call the default shell on any GNU/Linux system. I think this would work on mac, too. Definitely not Windows.